dali/pipeline/operators/reader/loader/video_loader.cc

```diff
@@ -175,40 +175,34 @@ OpenFile& VideoLoader::get_or_open_file(const std::string &filename) {
 
 
     auto stream = file.fmt_ctx_->streams[file.vid_stream_idx_];
+    int width = codecpar(stream)->width;
+    int height = codecpar(stream)->height;
     auto codec_id = codecpar(stream)->codec_id;
-    if (width_ == 0) {  // first file to open
-      width_ = codecpar(stream)->width;
-      height_ = codecpar(stream)->height;
+    if (max_width_ == 0) {  // first file to open
+      max_width_ = width;
+      max_height_ = height;
       codec_id_ = codec_id;
 
-      if (vid_decoder_) {
-        DALI_FAIL("Width and height not set, but we have a decoder?");
-      }
-      LOG_LINE << "Opened the first file, creating a video decoder" << std::endl;
-
-      vid_decoder_ = std::unique_ptr<NvDecoder>{
-          new NvDecoder(device_id_,
-                        codecpar(stream),
-                        stream->time_base,
-                        image_type_,
-                        dtype_,
-                        normalized_)};
     } else {  // already opened a file
-      if (!vid_decoder_) {
-          DALI_FAIL("width is already set but we don't have a vid_decoder_");
-      }
+      DALI_ENFORCE(codec_id_ == codec_id, "File " + filename +
+                   " is not the same codec as previous files");
+
+      if (NVCUVID_API_EXISTS(cuvidReconfigureDecoder)) {
+        if (max_width_ < width) max_width_ = width;
+        if (max_height_ < height) max_height_ = height;
 
-      if (width_ != codecpar(stream)->width ||
-          height_ != codecpar(stream)->height ||
-          codec_id_ != codec_id) {
-          std::stringstream err;
-          err << "File " << filename << " is not the same size and codec as previous files."
-              << " This is not yet supported. ("
-              << codecpar(stream)->width << "x" << codecpar(stream)->height
-              << " instead of "
-              << width_ << "x" << height_ << " or codec "
-              << codec_id << " != " << codec_id_ << ")";
-          DALI_FAIL(err.str());
+      } else {
+        if (max_width_ != width ||
+            max_height_ != height) {
+            std::stringstream err;
+            err << "File " << filename << " does not have the same resolution as previous files. ("
+                << width << "x" << height
+                << " instead of "
+                << max_width_ << "x" << max_height_ << "). "
+                << "Install Nvidia driver version >=396 (x86) or >=415 (Power PC) to decode"
+                   " multiple resolutions";
+            DALI_FAIL(err.str());
```

Comment:
Maybe you can put DALI_WARN and throw `unsupported_exception` and catch it in the test (and skip the case based on that).